idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/intentions/ElvisToIfThenIntention.kt

```diff
@@ -77,6 +77,18 @@ class ElvisToIfThenIntention : SelfTargetingRangeIntention<KtBinaryExpression>(K
         val left = KtPsiUtil.safeDeparenthesize(element.left!!)
         val right = KtPsiUtil.safeDeparenthesize(element.right!!)
 
+        val property = (KtPsiUtil.safeDeparenthesize(element).parent as? KtProperty)
+        val propertyName = property?.name
+        if (propertyName != null && right is KtReturnExpression) {
+            val parent = property.parent
+            val factory = KtPsiFactory(element)
+            factory.createExpressionByPattern("if ($0 == null) $1", propertyName, right)
+            parent.addAfter(factory.createExpressionByPattern("if ($0 == null) $1", propertyName, right), property)
+            parent.addAfter(factory.createNewLine(), property)
+            element.replace(left)
+            return
+        }
+
```

Comment:
Could you please also check how it works in safe cast situation (see code about safe cast receiver below)? For example
```
class My(val x: Int)
fun foo(arg: Any?) {
    val y = (arg as? My)?.x ?: return
    // Before your commit, it gives "val y = if (arg is My) arg.x else return", what after?
}
```